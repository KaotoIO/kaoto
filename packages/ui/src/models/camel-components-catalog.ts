import { CatalogKind } from './catalog-kind';

export interface ICamelComponentDefinition {
  component: ICamelComponent;
  componentProperties: ICamelComponentProperties;
  headers: ICamelComponentHeaders;
  properties: ICamelComponentProperties;
}

export interface ICamelComponent {
  kind: CatalogKind.Component;
  name: string;
  title: string;
  description: string;
  deprecated: boolean;
  firstVersion?: string;
  label: string;
  javaType?: string;
  supportLevel?: string;
  groupId?: string;
  artifactId?: string;
  version: string;
  scheme?: string;
  extendsScheme?: string;
  syntax?: string;
  async?: boolean;
  api?: boolean;
  consumerOnly?: boolean;
  producerOnly?: boolean;
  lenientProperties?: boolean;
}

export interface CatalogItemCommon {
  index: number;
  kind: string;
  displayName: string;
  group: string;
  label: string;
  required: boolean;
  javaType: string;
  deprecated: boolean;
  deprecationNote: string;
  autowired: boolean;
  secret: boolean;
  description: string;
}

export interface ICamelComponentProperties {
  type: string;
}

export interface ICamelComponentHeaders extends CatalogItemCommon {
  constantName: string;
}
