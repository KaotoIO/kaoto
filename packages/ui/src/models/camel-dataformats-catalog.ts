import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';
import { KaotoSchemaDefinition } from './kaoto-schema';

export interface ICamelDataformatDefinition {
  model: ICamelDataformatModel;
  properties: Record<string, ICamelDataformatProperty>;
  propertiesSchema: KaotoSchemaDefinition['schema'];
}

export type ICamelDataformatModel = ICamelProcessorModel;

export type ICamelDataformatProperty = ICamelProcessorProperty;
