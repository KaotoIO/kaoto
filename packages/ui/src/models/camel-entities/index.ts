export * from './base-entity';

/** Camel Route definitions */
export * from './flows/route';
