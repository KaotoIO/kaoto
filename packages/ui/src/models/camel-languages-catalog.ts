import { JSONSchemaType } from 'ajv';
import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';

export interface ICamelLanguageDefinition {
  model: ICamelLanguageModel;
  properties: Record<string, ICamelLanguageProperty>;
  propertiesSchema: JSONSchemaType<unknown>;
}

export interface ICamelLanguageModel extends ICamelProcessorModel {}

export interface ICamelLanguageProperty extends ICamelProcessorProperty {}
