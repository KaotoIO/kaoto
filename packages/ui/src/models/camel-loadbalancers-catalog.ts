import { JSONSchemaType } from 'ajv';
import { ICamelProcessorModel, ICamelProcessorProperty } from './camel-processors-catalog';

export interface ICamelLoadBalancerDefinition {
  model: ICamelLoadBalancerModel;
  properties: Record<string, ICamelLoadBalancerProperty>;
  propertiesSchema: JSONSchemaType<unknown>;
}

export interface ICamelLoadBalancerModel extends ICamelProcessorModel {}

export interface ICamelLoadBalancerProperty extends ICamelProcessorProperty {}
