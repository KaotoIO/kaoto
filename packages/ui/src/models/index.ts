export * from './camel-catalog-index';
export * from './camel-components-catalog';
export * from './camel-processors-catalog';
export * from './catalog-kind';
export * from './flow';
export * from './kamelets-catalog';
export * from './schema';
