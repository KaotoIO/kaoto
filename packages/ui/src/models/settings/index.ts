export * from './default-settings-adapter';
export * from './settings.model';
