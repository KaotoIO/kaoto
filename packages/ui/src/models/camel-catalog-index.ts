import { ICamelComponentDefinition } from './camel-components-catalog';
import { ICamelProcessorDefinition } from './camel-processors-catalog';
import { CatalogKind } from './catalog-kind';
import { IKameletDefinition } from './kamelets-catalog';

export interface CamelCatalogIndex {
  catalogs: Catalogs;
  schemas: CatalogEntry[];
}

export interface Catalogs {
  models: CatalogEntry;
  components: CatalogEntry;
  languages: CatalogEntry;
  dataformats: CatalogEntry;
  kamelets: CatalogEntry;
}

export interface CatalogEntry {
  name: string;
  version: string;
  file: string;
}

export type CatalogTypes = Record<string, ICamelComponentDefinition | ICamelProcessorDefinition | IKameletDefinition>;

export interface ComponentsCatalog {
  [CatalogKind.Component]?: Record<string, ICamelComponentDefinition>;
  [CatalogKind.Processor]?: Record<string, ICamelProcessorDefinition>;
  [CatalogKind.Kamelet]?: Record<string, IKameletDefinition>;
  [key: string]: unknown;
}

export interface CatalogCamelComponent {
  type: CatalogKind.Component;
  definition: ICamelComponentDefinition;
}

export interface CatalogCamelProcessor {
  type: CatalogKind.Processor;
  definition: ICamelProcessorDefinition;
}

export interface CatalogKamelet {
  type: CatalogKind.Kamelet;
  definition: IKameletDefinition;
}
