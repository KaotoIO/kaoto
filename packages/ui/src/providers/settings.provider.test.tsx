import { act, render } from '@testing-library/react';
import { useContext, useEffect, useState } from 'react';
import { DefaultSettingsAdapter, SettingsModel } from '../models/settings';
import { SettingsContext, SettingsProvider } from './settings.provider';

describe('SettingsProvider', () => {
  it('should render', () => {
    const settingsAdapter = new DefaultSettingsAdapter();

    const wrapper = render(
      <SettingsProvider adapter={settingsAdapter}>
        <TestProvider />
      </SettingsProvider>,
    );

    act(() => {
      expect(wrapper.getByTestId('settings')).toMatchSnapshot();
    });
  });
});

function TestProvider() {
  const settingsContext = useContext(SettingsContext);
  const [settings, setSettings] = useState<SettingsModel | null>(null);

  useEffect(() => {
    setSettings(settingsContext.getSettings());
  }, [settingsContext]);

  return <p data-testid="settings">{JSON.stringify(settings)}</p>;
}
