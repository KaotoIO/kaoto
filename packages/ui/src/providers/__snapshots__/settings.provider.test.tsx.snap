// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsProvider should render 1`] = `
<p
  data-testid="settings"
>
  {"catalogUrl":"","nodeLabel":"description"}
</p>
`;
